Different messages

tigobi opened the pull request.

tigobi committed:
Промежуточная версия с множеством спам-сообщений

tigobi committed:
Промежуточная версия с множеством спам-сообщений

tigobi committed:
Промежуточная версия
1 работает, но много - нет, надо фиксить(завтра)

tigobi committed:
Промежуточная версия
1 работает, но много - нет, теперь наверное у каждого свои таски(пожалуйста!!)

laSinteZ commented on src/main/java/org/tigobi/plugin/AdSpammer.java:
```diff
@@ -8,26 +8,34 @@
         this.plugin = plugin;
     }
+
+    private int playerId = 0;
```
Зачем?

laSinteZ commented on src/main/java/org/tigobi/plugin/AdSpammer.java:
```diff
@@ -8,26 +8,34 @@
             commandSender.sendMessage("adSpammer stop");
-            positionTask.cancel();
-            positionTask = null;
+            messageTasks.get(playerId).cancel();
```
а не свалится если таски нет? Сделай проверку на нулл

tigobi committed:
сейв версия
1 работает, но много - нет, теперь у каждого своя и останавливать только свою может.

laSinteZ commented on src/main/java/org/tigobi/plugin/AdSpammer.java:
```diff
@@ -41,12 +47,15 @@ private String argsToStringFrom2Element(String[] args) {
     }
 
-    private void whenStarts(String spamString) {
+    private HashMap<Integer, BukkitTask> whenStarts(String spamString, int playerId, HashMap<Integer, BukkitTask> tasks) {
```
1. Переименуй в что-то человеческое. что за whenStarts вообще. Что делает эта функция?
2. Зачем ты передаёшь в неё и возвращаешь из неё хешмапу? У тебя хешмапа – поле в классе, оттуда её и бери. Или вообще не трогай здесь хешмапу, ПРОСТО возвращай positionTask и снаружи функции клади его в мапу.
3. Есть ли научное объяснение зачем это вообще функция сейчас? Она вызывается ровно 1 раз.

laSinteZ commented on src/main/java/org/tigobi/plugin/AdSpammer.java:
```diff
@@ -41,12 +47,15 @@ private String argsToStringFrom2Element(String[] args) {
 
-    private void whenStarts(String spamString) {
+    private HashMap<Integer, BukkitTask> whenStarts(String spamString, int playerId, HashMap<Integer, BukkitTask> tasks) {
+        BukkitTask positionTask;
```
Зачем это в 2 строки. И используй var, зачем страдать.

laSinteZ commented on src/main/resources/plugin.yml:
```diff
@@ -1,5 +1,5 @@
 name: HelloWorldPlugin # the plugins name as it should appear in the plugin list /pl
```
снеси ты эти комментарии из файла, бесит же)))

laSinteZ reviewed (changes requested):
Неплохо-неплохо в целом :)

tigobi committed:
Исправил всё по комментам

tigobi committed:
Исправил всё по комментам 2
